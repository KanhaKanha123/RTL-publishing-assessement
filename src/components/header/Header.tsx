import { Container, Navbar, Nav, Dropdown, Badge } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const Header = () => {

    return <Navbar data-testid="header-wrapper-div" className='header'>
        <Container >
            <Navbar.Brand>
                <Link className='headerText' data-testid="header-anchor" to='/'>RTL News</Link>
            </Navbar.Brand>
            <Nav>
                <Dropdown>
                    <Badge className='headerText' data-testid="header-login" bg="inherit">Login</Badge>
                </Dropdown>
            </Nav>
        </Container>
    </Navbar>
}

export default Header;