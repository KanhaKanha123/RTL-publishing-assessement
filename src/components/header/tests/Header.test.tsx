import Header from '../Header';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Header component test', () => {

    test("Header component load correctly", () => {
        render(<Router><Header /></Router>);
        const wrapper = screen.getByTestId("header-wrapper-div");
        const headerAnchor = screen.getByText(/RTL News/i);
        const headerLoginText = screen.getByText(/Login/i);

        //check element exist
        expect(wrapper).toBeInTheDocument();
        expect(headerAnchor).toBeInTheDocument();
        expect(headerLoginText).toBeInTheDocument();
    });

    test("Header component should match snapshot", () => {
        const element = renderer.create(<Router><Header /></Router>);
        expect(element.toJSON()).toMatchSnapshot();
    });
})
