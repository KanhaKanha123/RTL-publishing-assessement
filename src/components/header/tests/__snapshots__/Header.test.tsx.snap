// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test Header component should match snapshot 1`] = `
<nav
  className="header navbar navbar-expand navbar-light"
  data-testid="header-wrapper-div"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <a
        className="headerText"
        data-testid="header-anchor"
        href="/"
        onClick={[Function]}
      >
        RTL News
      </a>
    </span>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="dropdown"
      >
        <span
          className="headerText badge bg-inherit"
          data-testid="header-login"
        >
          Login
        </span>
      </div>
    </div>
  </div>
</nav>
`;
